internal/gatewayapi/contraints.go

```diff
@@ -18,5 +18,13 @@ type RouteT interface {
 
 type BackendRefT interface {
 	BackendRef |
-		SecretObjectReference
+		SecretObjectReference |
+		// TODO TRR the plugin isn't really a backend in this sense, so that's a bit messy
+		// Do we need to distinguish between backend and other types of reference? at what level?
```

Comment:
My memory might fail me but as far as I recall this was only here to contrain the allowed types here.

Given that this only needs `Namespace`, `Name`, `Kind` and `Group` in `IsRefAllowedByGrant` you could remove all those type and replace the type constraint with `client.Object`. Do we want that? 🤷 It's up to us at what level we want to constrain the API.

Reply:
Alright, so basically no reason to not make it more or less generic for anything with a GVK and NamespacedName AFAICT. Re

> Nit: moving this in a separate PR would make the review a bit more focused

in `internal/gatewayapi/references.go`: at present this is mostly just moving what was previously in `internal/dataplane/translator/wrappers_refchecker.go` (since it's now used across packages, and would have caused an import cycle at its old location) aside the new case for plugins. How do you feel about a TODO issue to push that refactor to a separate PR?

If I _don't_ do that refactor it's just a relocation and a case addition in this PR, with the expectation that the cases will all collapse after refactoring.

Reply:
I have noticed that. My suggestion was just to make the PR more targeted so that the reviews are more focused, PRs smaller and then potential/hypothetical backports require less effort.

I'm OK with leaving this as is.